ffsync/memory_locker.go

```diff
@@ -81,27 +82,31 @@ func (m *memoryLocker) isPrefixOfExistingKey(key string) (string, bool) {
 	existingKey := ""
 	isPrefix := false
 
-	for k := range m.lockedItems {
-		if strings.HasPrefix(k, key) {
+	m.lockedItems.Range(func(k, v interface{}) bool {
```

Comment:
does this work?

if the existing key is 123456 and the key is 123 will this fail?

Reply:
@ahmadnazeri